helm/opencensus-collector-app/templates/configmap.yaml

```diff
@@ -8,21 +8,29 @@ metadata:
 data:
   oc-collector-config: |
     receivers:
+{{- if .Values.enableOpencensusReceiver }}
       opencensus:
-        port: 55678
+        port: {{ .Values.ports.opencensusReceiver }}
         keepalive:
           server-parameters:
             max-connection-age: 120s
-            max-connection-age-grace: 30s            
+            max-connection-age-grace: 30s
+{{- end }}
+{{- if .Values.enableZipkinReceiver }}
       zipkin:
-        port: 9411
-      jaeger: {}
+        port: {{ .Values.ports.zipkinReceiver }}
```

Comment:
just curious: Why don't you name `port` more specifically similar to `jaeger-thrift-tchannel-port`, e.g. `zipkin-port`?

Reply:
Just to make the config more structured. You could do this differently, ie. organize everything around different ingest protocols.